From d6b23151a7fdecece554d1a445480212222af65e Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:21:43 -0400
Subject: [cpufreq] Don't pass CPU to cpufreq_add_dev_{symlink|interface}()

Message-id: <1409235879-1283-10-git-send-email-prarit@redhat.com>
Patchwork-id: 89386
O-Subject: [RHEL7.1 PATCH BZ 1134639 009/185] cpufreq: Don't pass CPU to cpufreq_add_dev_{symlink|interface}()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 308b60e71541518f3fe97171b4daf71adc607f3d
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Wed Jul 31 14:35:14 2013 +0200

    cpufreq: Don't pass CPU to cpufreq_add_dev_{symlink|interface}()

    Pointer to struct cpufreq_policy is already passed to these routines
    and we don't need to send policy->cpu to them as well.  So, get rid
    of this extra argument and use policy->cpu everywhere.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 070bda0..a455490 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -781,8 +781,7 @@ void cpufreq_sysfs_remove_file(const struct attribute *attr)
 EXPORT_SYMBOL(cpufreq_sysfs_remove_file);
 
 /* symlink affected CPUs */
-static int cpufreq_add_dev_symlink(unsigned int cpu,
-       struct cpufreq_policy *policy)
+static int cpufreq_add_dev_symlink(struct cpufreq_policy *policy)
 {
  unsigned int j;
  int ret = 0;
@@ -790,10 +789,11 @@ static int cpufreq_add_dev_symlink(unsigned int cpu,
  for_each_cpu(j, policy->cpus) {
   struct device *cpu_dev;
 
-  if (j == cpu)
+  if (j == policy->cpu)
    continue;
 
   pr_debug("Adding link for CPU: %u\n", j);
+  cpufreq_cpu_get(policy->cpu);
   cpu_dev = get_cpu_device(j);
   ret = sysfs_create_link(&cpu_dev->kobj, &policy->kobj,
      "cpufreq");
@@ -803,8 +803,7 @@ static int cpufreq_add_dev_symlink(unsigned int cpu,
  return ret;
 }
 
-static int cpufreq_add_dev_interface(unsigned int cpu,
-         struct cpufreq_policy *policy,
+static int cpufreq_add_dev_interface(struct cpufreq_policy *policy,
          struct device *dev)
 {
  struct freq_attr **drv_attr;
@@ -840,7 +839,7 @@ static int cpufreq_add_dev_interface(unsigned int cpu,
    goto err_out_kobj_put;
  }
 
- ret = cpufreq_add_dev_symlink(cpu, policy);
+ ret = cpufreq_add_dev_symlink(policy);
  if (ret)
   goto err_out_kobj_put;
 
@@ -1051,7 +1050,7 @@ static int __cpufreq_add_dev(struct device *dev, struct subsys_interface *sif,
  write_unlock_irqrestore(&cpufreq_driver_lock, flags);
 
  if (!frozen) {
-  ret = cpufreq_add_dev_interface(cpu, policy, dev);
+  ret = cpufreq_add_dev_interface(policy, dev);
   if (ret)
    goto err_out_unregister;
  }
-- 
1.7.1